src/steamship/agents/examples/my_assistant.py

```diff
@@ -26,18 +27,20 @@ def __init__(self, **kwargs):
                 SearchTool(),
                 DalleTool(),
             ],
-            llm=OpenAI(self.client),
+            llm=OpenAI(self.client, temperature=0),
+            conversation_memory=MessageWindowMemory(k=2),
         )
 
     @post("prompt")
-    def prompt(self, prompt: str) -> str:
+    def prompt(self, prompt: str, context_id: Optional[uuid.UUID] = None) -> str:
         """Run an agent with the provided text as the input."""
 
         # AgentContexts serve to allow the AgentService to run agents
         # with appropriate information about the desired tasking.
-        # Here, we create a new context on each prompt, and append the
-        # prompt to the message history stored in the context.
-        context_id = uuid.uuid4()
+        # Here, we use the passed in context (or a new context) for the prompt,
+        # and append the prompt to the message history stored in the context.
+        if not context_id:
+            context_id = uuid.uuid4()
         context = AgentContext.get_or_create(self.client, {"id": f"{context_id}"})
```

Comment:
Do we need to push this down somewhere? Feels like every assistant class will have to implement this bit.

Reply:
I think this might be tough to push down. Context IDs could depend on a number of factors. Finding a simplification for telegram bot use cases, however, does seem useful, if we can figure it out properly.